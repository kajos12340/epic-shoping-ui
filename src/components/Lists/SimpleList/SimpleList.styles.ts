import styled from 'styled-components';

export const Container = styled.section`
  padding: 1em;
  
  display: grid;
  grid-template: "logo date" auto
                 "logo title" 4rem
                 "navigation navigation" 3em
                 / auto 1fr;
  grid-column-gap: 20px;
`;

export const Icon = styled.div`
  grid-area: logo;

  display: flex;
  justify-content: center;
  align-items: center;

  width: 75px;
  height: 75px;
  background-color: #00796b;
  border-radius: 10px;
  
  & svg {
    color: white;
    font-size: 2em;
  }
`;

export const Date = styled.div`
  grid-area: date;
  font-size: 0.85em;
  font-weight: 500;
  color: #676767;
`;

export const Title = styled.div`
  grid-area: title;
`;

export const Navigation = styled.div`
  padding-top: 10px;
  grid-area: navigation;
  display: flex;
  justify-content: flex-end;
  align-items: center;
`;

export const ListInactive = styled.span`
  font-weight: 500;
  color: red;
`;
