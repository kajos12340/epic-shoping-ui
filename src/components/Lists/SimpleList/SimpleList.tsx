import React from 'react';
import ShoppingCartOutlinedIcon from '@material-ui/icons/ShoppingCartOutlined';
import Typography from '@material-ui/core/Typography';

import { Button } from '@material-ui/core';
import {
  Container, Icon, Date, Title, Navigation, ListState,
} from './SimpleList.styles';

export interface ISimpleListItem {
  name: string,
  date: string,
  isActive: boolean,
  productsNumber: number,
  id: string,
}

const SimpleList = ({
  name, date, isActive, productsNumber,
}: ISimpleListItem) => (
  <Container>
    <Icon>
      <ShoppingCartOutlinedIcon />
    </Icon>
    <Date>
      {date}
    </Date>
    <Title>
      <Typography variant="h6">{name}</Typography>
      <Typography variant="subtitle2">{`${productsNumber} produktów na liście`}</Typography>
    </Title>
    <Navigation>
      <ListState isActive={isActive}>
        {isActive ? 'Lista aktywna' : 'Lista zakończona'}
      </ListState>
      <Button
        variant="contained"
        color={isActive ? 'secondary' : 'inherit'}
      >
        {isActive ? 'Przejdź do listy' : 'Szczegóły'}
      </Button>
    </Navigation>
  </Container>
);

export default SimpleList;
