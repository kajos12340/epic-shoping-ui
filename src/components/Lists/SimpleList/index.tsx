import React from 'react';
import ShoppingCartOutlinedIcon from '@material-ui/icons/ShoppingCartOutlined';
import Typography from '@material-ui/core/Typography';
import { useHistory } from 'react-router-dom';

import { Button } from '@material-ui/core';
import {
  Container, Icon, Date, Title, Navigation, ListInactive,
} from './styles';

export interface ISimpleListItem {
  name: string,
  date: string,
  isActive: boolean,
  productsNumber: number,
  _id: string,
  author: {
    _id: string,
    login: string,
  },
}

const SimpleList = ({
  name, date, isActive, productsNumber, _id, author,
}: ISimpleListItem) => {
  const history = useHistory();

  const handleListRedirect = () => {
    history.push(`/shopping/list/${_id}`);
  };

  return (
    <Container>
      <Icon>
        <ShoppingCartOutlinedIcon />
      </Icon>
      <Date>
        {`${date} - ${author.login}`}
      </Date>
      <Title>
        <Typography variant="h6" style={{ wordBreak: 'break-word' }}>{name}</Typography>
        <Typography variant="subtitle2">{`${productsNumber} produktów na liście`}</Typography>
      </Title>
      <Navigation>
        {isActive
          ? (
            <Button
              variant="contained"
              color="secondary"
              onClick={handleListRedirect}
            >
              {isActive ? 'Przejdź do listy' : 'Szczegóły'}
            </Button>
          )
          : (
            <ListInactive>
              Lista nieaktywna
            </ListInactive>
          )}
      </Navigation>
    </Container>
  );
};

export default SimpleList;
