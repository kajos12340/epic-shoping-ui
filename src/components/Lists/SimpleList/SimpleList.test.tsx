import React from 'react';
import { mount } from 'enzyme';
import { Button } from '@material-ui/core';
import { BrowserRouter } from 'react-router-dom';

import SimpleList, { ISimpleListItem } from './index';

describe('SimpleList', () => {
  const propsBase = {
    name: 'test name',
    isActive: true,
    productsNumber: 45,
    date: '12:33 24.05.2020',
    _id: 'list_id',
    author: {
      _id: 'idd2',
      login: 'login',
    },
  };

  const createWrapper = ({
    name, isActive, productsNumber, date, _id, author,
  }: ISimpleListItem) => mount(
    <BrowserRouter>
      <SimpleList
        name={name}
        date={date}
        isActive={isActive}
        productsNumber={productsNumber}
        _id={_id}
        author={author}
      />
    </BrowserRouter>,
  );

  it('should match snapshot for active list', () => {
    const wrapper = createWrapper({
      ...propsBase,
      isActive: true,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot for inactive list', () => {
    const wrapper = createWrapper({
      ...propsBase,
      isActive: false,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should redirect to singleList', () => {
    const wrapper = createWrapper({
      ...propsBase,
      isActive: true,
    });
    // @ts-ignore
    global.window = { location: { pathname: '/shopping/lists' } };

    const button = wrapper.find(Button);
    button.simulate('click');

    expect(global.location.pathname).toBe('/shopping/list/list_id');
  });
});
