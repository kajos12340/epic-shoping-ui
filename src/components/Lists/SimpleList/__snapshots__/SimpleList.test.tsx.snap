// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleList should match snapshot for active list 1`] = `ReactWrapper {}`;

exports[`SimpleList should match snapshot for inactive list 1`] = `ReactWrapper {}`;
